### PR TITLE
UnsupportedOperationException from JDK8 252 onwards

### PR DESCRIPTION
**Pull request by yschimke**
Fix for https://github.com/square/okhttp/issues/5970

Activates Jdk9Platform on Java 8 when the new methods are present (builds 252+).  Also handles the UnsupportedOperationException.

### CODE REVIEW

**Comment by yschimke** [comment]
Raised by @DanielThomas we need to change activation logic here also.  Otherwise we won't work on 8-252

File: okhttp/src/main/kotlin/okhttp3/internal/platform/Jdk9Platform.kt [hunk 1]
```diff
@@ -40,11 +40,16 @@ open class Jdk9Platform : Platform() {
...
+    try {
+      // SSLSocket.getApplicationProtocol returns "" if application protocols values will not
+      // be used. Observed if you didn't specify SSLParameters.setApplicationProtocols
+      return when (val protocol = sslSocket.applicationProtocol) {
+        null, "" -> null
+        else -> protocol
+      }
+    } catch (e: UnsupportedOperationException) {
+      // https://docs.oracle.com/javase/9/docs/api/javax/net/ssl/SSLSocket.html#getApplicationProtocol--
+      return null
```
Comment (monkey-mas) [inline okhttp/src/main/kotlin/okhttp3/internal/platform/Jdk9Platform.kt:52]: Maybe we can add test in `Jdk9PlatformTest`?
```java
  @Test
  public void selectedProtocolIsNullWhenSslSocketThrowsExceptionForApplicationProtocol() {
    assertThat(new Jdk9Platform().getSelectedProtocol(
        new SSLSocket() {
          @Override public String getApplicationProtocol() {
            throw new UnsupportedOperationException("Mock exception");
          }

          // Implement abstract methods.
          @Override public String[] getSupportedCipherSuites() {
            return new String[0];
          }

          @Override public String[] getEnabledCipherSuites() {
            return new String[0];
          }

          @Override public void setEnabledCipherSuites(String[] suites) {

          }

          @Override public String[] getSupportedProtocols() {
            return new String[0];
          }

          @Override public String[] getEnabledProtocols() {
            return new String[0];
          }

          @Override public void setEnabledProtocols(String[] protocols) {

          }

          @Override public SSLSession getSession() {
            return null;
          }

          @Override public void addHandshakeCompletedListener(HandshakeCompletedListener listener) {

          }

          @Override
          public void removeHandshakeCompletedListener(HandshakeCompletedListener listener) {

          }

          @Override public void startHandshake() throws IOException {

          }

          @Override public void setUseClientMode(boolean mode) {

          }

          @Override public boolean getUseClientMode() {
            return false;
          }

          @Override public void setNeedClientAuth(boolean need) {

          }

          @Override public boolean getNeedClientAuth() {
            return false;
          }

          @Override public void setWantClientAuth(boolean want) {

          }

          @Override public boolean getWantClientAuth() {
            return false;
          }

          @Override public void setEnableSessionCreation(boolean flag) {

          }

          @Override public boolean getEnableSessionCreation() {
            return false;
          }
        })).isNull();
```

**Comment by monkey-mas** [comment]
@swankjesse Can you also take a look please?

File: okhttp/src/test/java/okhttp3/internal/platform/Jdk9PlatformTest.java [hunk 2]
```diff
@@ -15,22 +15,118 @@
...
   public void testToStringIsClassname() {
     assertThat(new Jdk9Platform().toString()).isEqualTo("Jdk9Platform");
   }
+
+  @Test
+  public void selectedProtocolIsNullWhenSslSocketThrowsExceptionForApplicationProtocol() {
+    platform.assumeJdk9();
+
+    assertThat(new Jdk9Platform().getSelectedProtocol(
+        new SSLSocket() {
```
Comment (swankjesse) [inline okhttp/src/test/java/okhttp3/internal/platform/Jdk9PlatformTest.java:61]: `new DelegatingSSLSocket(null)` ?

Comment (monkey-mas) [reply okhttp/src/test/java/okhttp3/internal/platform/Jdk9PlatformTest.java:61]: > new DelegatingSSLSocket(null) ?
Cool! Didn't realize that!

**Pull request merged**